examples/integrations/oci/objectstorage-reactive/src/main/java/io/helidon/examples/integrations/oci/objecstorage/reactive/ObjectStorageService.java

```diff
@@ -52,67 +72,153 @@ public void update(Routing.Rules rules) {
+
+        objectStorageAsyncClient.deleteObject(DeleteObjectRequest.builder()
+                                                      .namespaceName(namespaceName)
+                                                      .bucketName(bucketName)
+                                                      .objectName(objectName).build(), deleteObjectHandler);
+        try {
+            DeleteObjectResponse deleteObjectResponse = deleteObjectHandler.waitForCompletion();
+            res.status(Http.Status.OK_200)
+                    .send();
+            return;
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error deleting object", e);
+            res.status(Http.Status.INTERNAL_SERVER_ERROR_500).send();
+            return;
+        }
     }
 
     private void rename(ServerRequest req, ServerResponse res) {
         String oldName = req.path().param("old-name");
         String newName = req.path().param("new-name");
 
-        objectStorage.renameObject(RenameObject.Request.builder()
-                                           .bucket(bucketName)
-                                           .objectName(oldName)
-                                           .newObjectName(newName))
-                .forSingle(it -> res.send("Renamed to " + newName))
-                .exceptionally(res::send);
+        RenameObjectRequest renameObjectRequest = RenameObjectRequest.builder()
+                .namespaceName(namespaceName)
+                .bucketName(bucketName)
+                .renameObjectDetails(RenameObjectDetails.builder()
+                                             .newName(newName)
+                                             .sourceName(oldName)
+                                             .build())
+                .build();
+
+        ResponseHandler<RenameObjectRequest, RenameObjectResponse> renameObjectHandler =
+                new ResponseHandler<>();
+
+        try {
+            objectStorageAsyncClient.renameObject(renameObjectRequest, renameObjectHandler);
+            RenameObjectResponse renameObjectResponse = renameObjectHandler.waitForCompletion();
+            res.status(Http.Status.OK_200)
+                    .send();
+            return;
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error renaming object", e);
+            res.status(Http.Status.INTERNAL_SERVER_ERROR_500).send();
+            return;
+        }
     }
 
     private void upload(ServerRequest req, ServerResponse res) {
-        OptionalLong contentLength = req.headers().contentLength();
-        if (contentLength.isEmpty()) {
-            req.content().forEach(DataChunk::release);
-            res.status(Http.Status.BAD_REQUEST_400).send("Content length must be defined");
+        String objectName = req.path().param("file-name");
+        PutObjectRequest putObjectRequest = null;
+        try {
+            InputStream stream = new FileInputStream(System.getProperty("user.dir") + File.separator + objectName);
```

Comment:
This should be within try with resources, your are not closing the stream.
Please make sure this is done in all cases where streams are used

Reply:
Done and verified across all usages.